cli/gluster-block.c

```diff
@@ -1027,14 +1017,59 @@ glusterBlockParseArgs(int count, char **options)
 int
 main(int argc, char *argv[])
 {
+  size_t opt = 0;
+  int count = 1;
+  int args = argc;
+  int json = GB_JSON_NONE;
+
+
   if (argc <= 1) {
     glusterBlockHelp();
-    exit(EXIT_FAILURE);
+    goto fail;
   }
 
   if(initLogging()) {
-    exit(EXIT_FAILURE);
+    goto fail;
+  }
+
+  while (1) {
+    opt = glusterBlockCLIOptEnumParse(argv[count]);
+    if (opt < 1 || opt >= GB_CLI_OPT_MAX) {
+      MSG(stderr, "Unknown option: %s\n", argv[count]);
+      goto fail;
+    }
+    if (opt == GB_CLI_TIMEOUT) {
+      if (args <= 3) {     /* some command is manditory */
+        glusterBlockHelp();
+        goto fail;
+      }
+
+      if (isNumber(argv[2])) {
+        sscanf(argv[2], "%ld", &cliOptTimeout);
+      } else {
+        MSG(stderr, "%s\n", "'timeout' option is incorrect, check usage.");
+        MSG(stderr, "%s\n", "hint: timeout argument accept only time in seconds.");
+        goto fail;
+      }
+      printf ("option %s, value %s\n", argv[1], argv[2]);
```

Comment:
Is this needed here ?

Reply:
Except this all the others look good to me and also tested and works for me.

Reply:
Xiubo, thanks for catching this, have added this while writing the code for debug info. Will remove it and spin this version soon.